### DISCUSSION TITLE
C4GT: Learner Passbook integration with Sunbird ED Apps

### DISCUSSION THREAD

**Discussion by cchaitanya27** [Sunbird-Ed/Community#488] [Feature request]
### Project Details
Sunbird ED is a platform that can be used to deploy scale-ready solutions in the area of learning, capacity building, and professional development. The platform provides capabilities for a user to register and take trainings that are hosted on the platform. The trainings can also have certificates issued. Each learner has a profile that lists all the courses taken by him/her.

This is to extend to integrate Digital Learner Passbook with Sunbird Ed Portal and Mobile Apps. A digital learner passbook will act as a digital profile & docket of a learner wherein it will comprise details of the learner along with all the digitally verifiable credentials and will also capture the skills/competencies that have been achieved by the learner. The verifiable credentials can be issued by different sources, not just the Sunbird ED solution.

### Features to be implemented
Integrate the pluggable Digital Learner Passbook in the Profile page of Sunbird ED Portal App.
Integrate the pluggable Digital Learner Passbook in the Profile page of Sunbird ED Mobile App

### What exists
Sunbird ED Portal and Mobile Apps
Digital Learner Passbook Service

### What needs to be built
UI in Sunbird ED Portal and Mobile apps to display Digital Learner Passbook.

### Learning Path
Complexity: Medium
Skills Required: 
Understanding existing Sunbird ED, Sunbird Learn, Sunbird RC
NodeJS, Angular
Mentors: Niharika, Chinmay
Project size: 6 weeks

### Milestones
Detailing and documenting the capabilities
Creating the design of the Digital Learner's Passbook UI in Portal and Mobile
Implementation of the Digital Learner's Passbook UI
Developing and running Test cases

**Comment by chitranshu-keshav** [comment]
@Sunbird-Ed/sunbirded_workinggroup  Can you please review and provide your inputs on this feature request

**Reply by Ashish11828** [reply]
Hello Mentor, I like to contribute in this project how can we contribute in this project, looking forward to submitting the proposal

**Comment by Karan9034** [comment]
Hello mentors,
I would like to contribute to this project. Could you guide me on how to get started or provide any resources to go through? Looking forward to submitting a proposal for this project.

**Reply by kameshbhr** [reply]
Hi Karan, 
Glad to know you are interested in the project. Since this is a C4GT project, can you please post your queries in the C4GT discord channel?

**Reply by Karan9034** [reply]
Yeah, I have posted a message there as well.

**Comment by Mounika-Pavanthi** [comment]
I am thrilled to contribute to the Digital Learner Passbook project and be part of the team that enhances the capabilities of Sunbird ED. With my expertise in Java, Node, and Angular, combined with a deep understanding of Sunbird platforms, I am confident in my ability to develop the Digital Learner Passbook service. I am dedicated to delivering high-quality code, collaborating with mentors, and achieving project milestones within the allocated timeframe. Please consider my involvement in the project, as I am passionate about empowering learners through digital profiles and verifiable credentials.

**Reply by kameshbhr** [reply]
@Mounika-Pavanthi Great to hear Mounika! Looking forward to your proposal.

**Comment by adityakode** [comment]
I would love to work on this for C4GT.

**Comment by GVishnudhasan** [comment]
Hello Project Admin, 
I am super interested and very keen to work on this project. Previously I have worked on Angular projects and JavaScript backend. This particular feature request excites me a lot and I am very keen to work on this feature.

**Comment by dharmendra-byte** [comment]
Hello Sir @kameshbhr 
I am super interested and very keen to work on this project. Previously I have worked on projects and JavaScript backend. This particular feature request excites me a lot and I am very keen to work on this feature.
